### PR TITLE
Implementation of the Storecomponents npm module

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Implementation of the BuyButton component that calls the checkout api to add an item to the cart

#### What problem is this solving?
Fix issue #1 

#### How should this be manually tested?/

Available at:
https://buybutton--storecomponents.myvtex.com/

#### Screenshots or example usage

image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: src/BuyButton/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import addToCartMutation from './mutations/addToCartMutation.gql'
+
+/**
+ * BuyButton Component. Adds a list of itens to the cart.
+ */
+class BuyButton extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.handleAddToCart = this.handleAddToCart.bind(this)
```
Comment (ClaudivanFilho) [inline src/BuyButton/index.js]: Em vez de dar bind declara a função como uma arrow function. Vitória, Estácio e eu estamos fazendo assim, melhor seguir o padrão.

```handleAddToCart = () => { ```

File: src/BuyButton/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    } = this.props
+
+    mutate({
+      variables: {
+        orderFormId: orderFormId,
+        items: [
+          {
+            id: parseInt(skuId),
+            index: 1,
+            quantity: quantity,
```
Comment (ClaudivanFilho) [inline src/BuyButton/index.js]:
```javascript
index: 1,
quantity,  <------
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "react": "^15.5.4",
+    "react-apollo": "^2.1.3",
+    "webpack": "^2.6.1"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "webpack --watch",
+    "build": "webpack"
+  },
+  "author": "Vtex",
```
Comment (brunojdo) [inline package.json]: VTEX ⛏

File: .babelrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (brunojdo) [inline .babelrc]: create a `CHANGELOG.md` Follow this model https://github.com/vtex-apps/store-graphql/blob/master/CHANGELOG.md

File: src/BuyButton/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      <Button primary onClick={this.handleAddToCart}>
+        {this.props.children}
+      </Button>
+    )
+  }
+}
+
+BuyButton.propTypes = {
+  /** Message that will be displayed inside of the button **/
+  children: PropTypes.PropTypes.node.isRequired,
```
Comment (ClaudivanFilho) [inline src/BuyButton/index.js]: Não seria melhor passar como uma prop text em vez de uma children?

File: src/BuyButton/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  /** The quantity of products to be added to the cart */
+  quantity: PropTypes.number.isRequired,
+  /** The specification of wich product will be added to the cart */
+  skuId: PropTypes.number.isRequired,
+  /** Wich seller is being referenced by the button */
```
Comment (ClaudivanFilho) [inline src/BuyButton/index.js]: which

File: src/BuyButton/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    )
+  }
+}
+
+BuyButton.propTypes = {
+  /** Message that will be displayed inside of the button **/
+  children: PropTypes.PropTypes.node.isRequired,
+  /** The quantity of products to be added to the cart */
+  quantity: PropTypes.number.isRequired,
+  /** The specification of wich product will be added to the cart */
```
Comment (ClaudivanFilho) [inline src/BuyButton/index.js]: which

File: src/BuyButton/index.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  skuId: PropTypes.number.isRequired,
+  /** Wich seller is being referenced by the button */
+  seller: PropTypes.string.isRequired,
+  /** Channel */
+  salesChannel: PropTypes.string.isRequired,
+  /** Graphql property to call a mutation */
+  mutate: PropTypes.func.isRequired,
+  /** Function that will be called after the mutation */
+  afterClick: PropTypes.func.isRequired,
+  /** The users cart id */
```
Comment (ClaudivanFilho) [inline src/BuyButton/index.js]: user

File: README.md [hunk 9]
```diff
@@ -1,2 +1,61 @@
...
+
+```javascript
+/**
+ * The quantity of products to be added to the cart
+ */
+- quantity
+  - Type: Number
+
+/**
+ * The specification of wich product will be added to the cart.
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply README.md]: 👍

File: README.md [hunk 10]
```diff
@@ -1,2 +1,61 @@
...
+ */
+- skuId
+  - Type: Number
+
+/**
+ * Wich seller is being referenced by the button.
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply README.md]: 👍

File: README.md [hunk 11]
```diff
@@ -1,2 +1,61 @@
...
+  - Type: Function
+
+/**
+ * Function that will be called after the mutation.
+ */
+- afterClick
+  - Type: Function
+
+/**
+ * The users cart id
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply README.md]: 👍

File: .gitignore [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+# App specific
+node_modules/
+docs/_book/
+.tmp
+.idea
+npm-debug.log
+.build/
+lib
+build
+.eslintrc
```
Comment (klzns) [inline .gitignore]: Tirar do .gitignore

File: package.json [hunk 13]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "vtex.buybutton",
```
Comment (klzns) [inline package.json]: Trocar o nome do pacote para:

```js
@vtex/buy-button
```

File: docs/introduction.md [hunk 14]
```diff
@@ -0,0 +1,13 @@
+Oie! These is our buy button component component.
+
+<div class="pa5 br2 bg-washed-yellow">
+    This component still under construction. Everything may - and will - break. Use with caution, and avoid for now its use in Production projects.
+</div>
+
+## Install
+
+```sh
+yarn add @vtex/buybutton --exact
```
Comment (klzns) [inline docs/introduction.md]: Trocar para `@vtex/buy-button`

Tirar `--exact`

File: docs/introduction.md [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+# or
+npm install @vtex/buybutton --save-exact
```
Comment (klzns) [inline docs/introduction.md]: Tirar `--save-exact`

File: package.json [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+  "version": "0.0.2",
+  "description":
+    "Component used to call a mutation that adds one item to the cart",
+  "main": "build/index.js",
+  "peerDependencies": {
+    "react": "^15.5.4"
+  },
+  "files": ["lib/"],
+  "dependencies": {
+    "@vtex/styleguide": "^2.0.0-rc.34",
```
Comment (klzns) [inline package.json]: Tirar `^`

File: .gitignore (hunk 12 above)
Comment (salesfelipe) [reply .gitignore]: 👍

File: package.json (hunk 13 above)
Comment (salesfelipe) [reply package.json]: Como a gente decidiu juntar todos em um só repo, eu coloquei: @vtex/storecomponents/buy-button 👍

File: docs/introduction.md (hunk 14 above)
Comment (salesfelipe) [reply docs/introduction.md]: 👍

File: docs/introduction.md (hunk 15 above)
Comment (salesfelipe) [reply docs/introduction.md]: 👍

File: package.json (hunk 16 above)
Comment (salesfelipe) [reply package.json]: 👍

File: package.json [hunk 17]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "vtex.buybutton",
+  "name": "@vtex/storecomponents/buy-button",
```
Comment (klzns) [inline package.json]: Trocar para `@vtex/buy-button`

File: docs/introduction.md [hunk 18]
```diff
@@ -7,7 +7,7 @@ Oie! These is our buy button component component.
 ## Install
 
 ```sh
-yarn add @vtex/buybutton --exact
+yarn add @vtex/storecomponents/buy-button
 # or
-npm install @vtex/buybutton --save-exact
+npm install @vtex/storecomponents/buy-button
```
Comment (klzns) [inline docs/introduction.md]: Idem

File: README.md [hunk 19]
```diff
@@ -1,2 +1,61 @@
-# buybutton
-Buy Button npm module 
+# BuyButton
+
+BuyButton is a canonical component that any VTEX store can install.
```
Comment (brunojdo) [inline README.md]: Change the description. Put this `README` in buybutton folder.

File: package.json (hunk 17 above)
Comment (salesfelipe) [reply package.json]: 👍

File: docs/introduction.md (hunk 18 above)
Comment (salesfelipe) [reply docs/introduction.md]: 👍

**Pull request merged**